client/components/OnboardingWizard.jsx

```diff
@@ -0,0 +1,158 @@
+  });
+
+  const steps = [
+    { id: "name", required: true },
+    { id: "age", required: true },
+    { id: "goal", required: true },
+    { id: "gender", required: true },
+    { id: "occupation", required: false },
+    { id: "availability", required: true },
+    { id: "preference", required: false },
+    { id: "diet", required: true },
+    { id: "metrics", required: true },
+    { id: "phone", required: false },
+  ];
+
+  const updateFormData = (field, value) =>
+    setFormData((prev) => ({ ...prev, [field]: value }));
+
+  const nextStep = () => {
+    if (currentStep < steps.length - 1) setCurrentStep((prev) => prev + 1);
+  };
+
+  const prevStep = () => {
+    if (currentStep > 0) setCurrentStep((prev) => prev - 1);
+  };
+
+  const skipStep = () => {
+    if (!steps[currentStep].required) nextStep();
+  };
+
+  const navigate = useNavigate();
+
+  const handleSubmit = async () => {
+    console.log("Submitting formData:", formData);
+    const resp = await fetch(
+      `${import.meta.env.VITE_API_URL}/auth/complete-onboarding`,
+      { method: "POST", credentials: "include" }
+    );
+    if (!resp.ok) {
+      console.error("Failed to mark onboarding complete");
+      return;
+    }
+    navigate("/dashboard");
+  };
+
+  return (
+    <div className="wizard-container">
+      Complete your FitSync registration here
+      <ProgressBar
+        currentStep={currentStep}
+        totalSteps={steps.length}
+      />
+
+      {currentStep === 0 && (
+        <StepName
+          value={formData.name}
+          setValue={(val) => updateFormData("name", val)}
+        />
+      )}
+      {currentStep === 1 && (
+        <StepAge
+          value={formData.age}
+          setValue={(val) => updateFormData("age", val)}
+        />
+      )}
+      {currentStep === 2 && (
+        <StepGoal
+          value={formData.goal}
+          setValue={(val) => updateFormData("goal", val)}
+        />
+      )}
+      {currentStep === 3 && (
+        <StepGender
+          value={formData.gender}
+          setValue={(val) => updateFormData("gender", val)}
+        />
+      )}
+      {currentStep === 4 && (
+        <StepOccupation
+          value={formData.occupation}
+          setValue={(val) => updateFormData("occupation", val)}
+        />
+      )}
+      {currentStep === 5 && (
+        <StepAvailability
+          value={formData.availability}
+          setValue={(val) => updateFormData("availability", val)}
+        />
+      )}
+      {currentStep === 6 && (
+        <StepPreference
+          value={formData.preference}
+          setValue={(val) => updateFormData("preference", val)}
+        />
+      )}
+      {currentStep === 7 && (
+        <StepDiet
+          value={formData.diet}
+          setValue={(val) => updateFormData("diet", val)}
+        />
+      )}
+      {currentStep === 8 && (
+        <StepMetrics
+          value={formData.biometric}
+          setValue={(val) => updateFormData("biometric", val)}
```

Comment:
should this be metrics?
```suggestion
          value={formData.metrics}
          setValue={(val) => updateFormData("metrics", val)}
```